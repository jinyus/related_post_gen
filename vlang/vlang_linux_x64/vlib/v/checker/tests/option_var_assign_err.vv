module main

fn trying_to_change_non_opt_to_option(value int) ?int {
	mut result := 122

	if value == 1 {
		result = trying_to_change_non_opt_to_option(0)
	}
	return result
}

fn main() {
	println(trying_to_change_non_opt_to_option(0))
}
