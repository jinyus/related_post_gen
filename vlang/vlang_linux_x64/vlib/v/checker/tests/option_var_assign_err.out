vlib/v/checker/tests/option_var_assign_err.vv:7:12: error: cannot assign an Option value to a non-option variable
    5 | 
    6 |     if value == 1 {
    7 |         result = trying_to_change_non_opt_to_option(0)
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    8 |     }
    9 |     return result
